tests/asg/asg_test.go

```diff
@@ -0,0 +1,156 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	. "github.com/portworx/torpedo/tests"
+
+	// https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	scaleTimeout = 10 * time.Minute
+)
+
+func TestASG(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	var specReporters []Reporter
+	junitReporter := reporters.NewJUnitReporter("/testresults/junit_basic.xml")
+	specReporters = append(specReporters, junitReporter)
+	RunSpecsWithDefaultAndCustomReporters(t, "Torpedo : ASG", specReporters)
+}
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+})
+
+// This test performs basic test of scaling up and down the asg cluster
+var _ = Describe("{ClusterScaleUpDown}", func() {
+	It("has to validate that storage nodes are not lost during asg scaledown", func() {
+
+		autoNodeRecoveryTimeout := os.Getenv("ASG_AUTO_NODE_RECOVERY_TIMEOUT_MIN")
```

Comment:
remove the knowledge of this variable from torpedo.

Reply:
Hi @harsh-px Should we be defaulting to 10 mins of recovery time?